### PR TITLE
DT-708 wiring in endpoint to service layer for court hearings without court case

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
PR to expose endpoint for creation of court hearings without a court case.

### CODE REVIEW

File: src/test/java/net/syscon/elite/api/resource/impl/ResourceTest.java [hunk 1]
```diff
@@ -26,7 +25,6 @@
 import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
 import static org.springframework.core.ResolvableType.forType;
 
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
```
Comment (matt-crowson-moj) [inline src/test/java/net/syscon/elite/api/resource/impl/ResourceTest.java:29]: So I removed this to see what difference it made.  No negative impact that I could see.  Please shout if you think differently.

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_scheduleCourtHearing.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        final var response = testRestTemplate.exchange(
+                "/api/bookings/-1/prison-to-court-hearings",
+                HttpMethod.POST,
+                request,
+                new ParameterizedTypeReference<String>() {
+                });
+
+
+        assertThat(response.getStatusCode()).isEqualTo(CREATED);
+        assertThat(getBodyAsJsonContent(response)).extractingJsonPathNumberValue("$.id").isNotNull();
```
Comment (matt-crowson-moj) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_scheduleCourtHearing.java:37]: This value is populated via a sequence on the DB so cannot guarantee the version.

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_scheduleCourtCaseHearing.java [hunk 3]
```diff
@@ -25,17 +26,23 @@ public void schedules_court_hearing() {
...
-                "/api/bookings/-1/court-cases/-1/prison-to-court-hearings",
+                "/api/bookings/-2/court-cases/-2/prison-to-court-hearings",
                 HttpMethod.POST,
                 request,
                 new ParameterizedTypeReference<String>() {
                 });
 
-        assertThatJsonFileAndStatus(response, 201, "schedule_court_hearing.json");
+        assertThat(response.getStatusCode()).isEqualTo(CREATED);
+        assertThat(getBodyAsJsonContent(response)).extractingJsonPathNumberValue("$.id").isNotNull();
```
Comment (matt-crowson-moj) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_scheduleCourtCaseHearing.java:36]: This value is populated via a sequence on the DB so cannot guarantee the version.

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import org.springframework.http.HttpMethod;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class OffenderMovementsResourceImplIntTest_getCourtHearings extends ResourceTest {
+
+    @Test
+    public void get_court_hearings_for_booking_returns_no_court_hearings() {
```
Comment (JamieDraperUK) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java:17]: Ruby? :trollface:

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        final var request = createHttpEntity(token, null);
+
+        final var response = testRestTemplate.exchange(
+                "/api/bookings/-3/court-hearings",
+                HttpMethod.GET,
+                request,
+                new ParameterizedTypeReference<String>() {
+                });
+
+        assertThatJsonFileAndStatus(response, 200, "get_court_hearings_3_for_booking.json");
```
Comment (brightonsbox) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java:45]: Whati is this 3 relating to? The booking id or the number of court hearings?

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_scheduleCourtCaseHearing.java [hunk 6]
```diff
@@ -45,7 +52,7 @@ public void schedule_court_hearing_fails_when_no_matching_booking() {
         ));
 
         final var response = testRestTemplate.exchange(
-                "/api/bookings/9999999/court-cases/-1/prison-to-court-hearings",
+                "/api/bookings/9999999/court-cases/-2/prison-to-court-hearings",
```
Comment (brightonsbox) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_scheduleCourtCaseHearing.java:55]: Did this need to be changed?

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java (hunk 4 above)
Comment (matt-crowson-moj) [reply src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java:17]: Someone noticed...I have to be honest I prefer this to Java convention. https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
+package net.syscon.elite.api.resource.impl;
+
+import net.syscon.elite.api.model.ErrorResponse;
+import net.syscon.elite.executablespecification.steps.AuthTokenHelper;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+
+import java.util.Map;
```
Comment (mikehalmamoj) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java:10]: For a future PR - there are a couple of unused imports.

Comment (matt-crowson-moj) [reply src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest_getCourtHearings.java:10]: Nah. I'll sort.  Missed a test rename anyway.

**Pull request merged**